### PR TITLE
Cleanup InputLanguage interop

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup InputLanguage interop

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2321)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs [hunk 1]
```diff
@@ -5,13 +5,14 @@
...
 using System.Linq;
+using System.Reflection;
 using Xunit;
 
 namespace System.Windows.Forms.Tests
 {
     public class InputLanguageTests
     {
-        [Fact]
+        [WinFormsFact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs]: I don't think these should be `WinFormsFact`s, `InputLanguage` is not a control nor requires a UI thread

File: src/Common/src/Interop/User32/Interop.ActivateKeyboardLayout.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using System;
+using System.Runtime.InteropServices;
+using System.Windows.Forms;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [DllImport(Libraries.User32, ExactSpelling = true)]
+        public static extern IntPtr ActivateKeyboardLayout(IntPtr hkl, KLF Flags);
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.ActivateKeyboardLayout.cs]: We don't seem to be utilising `KLF` anywhere? Do we _really_ need to introduce it?

Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.ActivateKeyboardLayout.cs]: Yeah tbh. Removed

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs]: Fixe

File: src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs [hunk 3]
```diff
@@ -14,22 +14,22 @@ namespace System.Windows.Forms
     /// <summary>
     ///  Provides methods and fields to manage the input language.
     /// </summary>
-    public sealed class InputLanguage
+    public sealed class InputLanguage : IHandle
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs]: `IHandle` isn't needed here as we have no finalizer.

File: src/Common/src/Interop/User32/Interop.ActivateKeyboardLayout.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using System.Windows.Forms;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [DllImport(Libraries.User32, ExactSpelling = true)]
+        public static extern IntPtr ActivateKeyboardLayout(IntPtr hkl, uint Flags);
+
+        public static IntPtr ActivateKeyboardLayout(IHandle hkl, uint Flags)
```
Comment (JeremyKuhne) [inline src/Common/src/Interop/User32/Interop.ActivateKeyboardLayout.cs]: I don't think any of the callers have finalizers for the handle, so this isn't necessary.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs [hunk 5]
```diff
@@ -97,6 +105,13 @@ public void InputLanguage_FromCulture_NullCulture_ThrowsArgumentNullException()
             Assert.Throws<ArgumentNullException>("culture", () => InputLanguage.FromCulture(null));
         }
 
+        [Fact]
+        public void InputLanguage_GetHashCode_Invoke_ReturnsExpected()
+        {
+            InputLanguage language = InputLanguage.CurrentInputLanguage;
+            Assert.Equal(language.GetHashCode(), language.GetHashCode());
```
Comment (JuditRose) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs]: These two seems to be equal in all cases (if language is not null, then it's a null reference ex). Assert not null maybe? or what is expected?

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs]: This is deliberate - we should always return the same hash code from `GetHashCode` when called multiple times

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs]: @JuditRose I almost asked the same question. Then I looked at the implementation of `GetHashCode`:
https://github.com/dotnet/winforms/blob/3b37191254ad5f407fc9b1ea8643fe7cacf69924/src/System.Windows.Forms/src/System/Windows/Forms/InputLanguage.cs#L344-L347

Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/InputLanguageTests.cs]: @hughbe I suggest renaming the method to something like `InputLanguage_GetHashCode_RemainsSameAcrossCalls`

**Comment by RussKie** [comment]
MC

**Pull request merged**